### ISSUE TITLE
Support for Android SDK 31 - specify an explicit value for android:exported

### ISSUE THREAD

**Issue by salvatore-cirasa** [dpa99c/cordova-plugin-firebasex#684]
# Bug report

**CHECKLIST**
- [x] I have read the [issue reporting guidelines](https://github.com/dpa99c/cordova-plugin-firebasex#reporting-issues)

- [x] I confirm this is a suspected bug or issue that will affect other users

- [x] I have reproduced the issue using the [example project](https://github.com/dpa99c/cordova-plugin-firebasex-test) or provided the necessary information to reproduce the issue.

- [x] I have read [the documentation](https://github.com/dpa99c/cordova-plugin-firebasex) thoroughly and it does not help solve my issue.

- [x] I have checked that no similar issues (open or closed) already exist.

**Current behavior:**

When I try to build my app using SDK 31, the build fails with an error about a missing  value for `android:exported` in *AndroidManifest.xml*.

**Expected behavior:**

To build the app like what it is doing in previous SDK versions.

**Steps to reproduce:**

- Update targetSdkVersion to 31;
- Execute `cordova build android`;

**Screenshots**

**Environment information**

- Cordova CLI version 
	- `cordova -v`: 10.0.0
- Cordova platform version
	- `cordova platform ls`: android 10.1.2-dev
- Plugins & versions installed in project (including this plugin)
    - `cordova plugin ls`:
        - cordova-plugin-androidx-adapter 1.1.3 "cordova-plugin-androidx-adapter"
        - cordova-plugin-appinfo 2.1.2 "AppInfo Plugin"
        - cordova-plugin-appversion 1.0.0 "App Version"
        - cordova-plugin-background-fetch 7.0.3 "CDVBackgroundFetch"
        - cordova-plugin-buildinfo 4.0.0 "BuildInfo"
        - cordova-plugin-camera 6.0.0 "Camera"
        - cordova-plugin-device 2.0.3 "Device"
        - cordova-plugin-geolocation 4.1.0 "Geolocation"
        - cordova-plugin-inappbrowser 5.0.0 "InAppBrowser"
        - cordova-plugin-ionic-keyboard 2.2.0 "cordova-plugin-ionic-keyboard"
        - cordova-plugin-ionic-webview 4.2.1 "cordova-plugin-ionic-webview"
        - cordova-plugin-network-information 3.0.0 "Network Information"
        - cordova-plugin-screen-orientation 3.0.2 "Screen Orientation"
        - cordova-plugin-splashscreen 6.0.0 "Splashscreen"
        - cordova-plugin-statusbar 2.4.2 "StatusBar"
        - cordova-plugin-vibration 3.1.1 "Vibration"

- Dev machine OS and version, e.g.
    - Arch Linux 5.14.16-zen1-1-zen
        
_Android build issue:_	
- Node JS version
    - `node -v`: v16.11.1
- Gradle version
	- `ls platforms/android/.gradle`:
	   - 7.1.1
	   - 7.2
	   - buildOutputCleanup
	   - checksums
	   - vcs-1
- Target Android SDK version
	- `android:targetSdkVersion` in `AndroidManifest.xml`: 31
- Android SDK details
	- `sdkmanager --list | sed -e '/Available Packages/q'`
	
**Related code:**
For test and debug on my project, I modified `plugins/cordova-plugin-firebasex/plugin.xml` adding `android:exported="false"` to line 39. Next build ended with success.
```
<service android:exported="false" android:name="org.apache.cordova.firebase.FirebasePluginMessagingService">
    <intent-filter>
	    <action android:name="com.google.firebase.MESSAGING_EVENT"/>
    </intent-filter>
</service>
```

**Console output**
<details>
<summary>console output</summary>

```
> Task :app:processDebugMainManifest FAILED
~/myApp/platforms/android/app/src/main/AndroidManifest.xml Error:
        Apps targeting Android 12 and higher are required to specify an explicit value for `android:exported` when the corresponding component has an intent filter defined. See https://developer.android.com/guide/topics/manifest/activity-element#exported for details.
```

</details><br/><br/>

**Other information:**

**Comment by ossraju** [comment]
I also have same is there any  issue any work around ??

**Comment by salvatore-cirasa** [comment]
I found a workaround adding `android:exported="false"` to plugin.xml in every plugin that declare `<intent-filter>`, line 39 in this plugin. I don't know if this is the correct solution for production.
`cordova-android` plugin has this bugfix in master branch.

**Comment by yashp98** [comment]
i also faced same issue

**Comment by orenagiv** [comment]
Same here.

**Comment by eddragon** [comment]
First of all, thank you @dpa99c for the amazing plugin, it regularly makes my life a lot easier. 🙂

As for the issue at hand, @salvatore-cirasa is correct, a fix for it is as simple as that:

https://github.com/EpicDragonLtd/cordova-plugin-firebasex/commit/17edd2da4c2b09747aeeb7906dd412335374fd3c

It would be really great to see it in the main repo at some point!

**Comment by dpa99c** [comment]
Resolved by #702 which will be included in the upcoming next release

**Comment by alexwilson-motasoft** [comment]
#702 is missing from the 14.0.0-cli build, is this an oversight or should we be using something pre-build tool to switch the package version for Android/IOS CLI builds?

**Comment by nvahalik** [comment]
@alexwilson-motasoft The `-cli` variant of this package *does not* includes the fix, but the regular one does.

**Comment by ovidiugabriel** [comment]
The fix  17edd2d is present in cordova-plugin-firebasex version 14.2.1 (and the fix seems to be correct) but if I add cordova-plugin-firebasex to the build I get _"Apps targeting Android 12 and higher are required to specify an explicit value for `android:exported` when the corresponding component has an intent filter defined."_ 

When I remove cordova-plugin-firebasex the build is passing. It means that there is a dependency to cordova-plugin-firebasex that haven't fixed this already? How can I debug which npm dependency is exactly causing the issue?
